kart/subprocess_util.py

```diff
@@ -138,3 +141,46 @@ def subprocess_tee(cmd, **kwargs):
         read_and_display(cmd, **kwargs)
     )
     return return_code, stdout, stderr
+
+
+def run_with_capture(cmd, args, env):
+    # In testing, .run must be set to capture_output and so use PIPEs to communicate
+    # with the process to run whereas in normal operation the standard streams of
+    # this process are passed into subprocess.run.
+    # Capturing the output in a PIPE and then writing to sys.stdout is compatible
+    # with click.testing which sets sys.stdout and sys.stderr to a custom
+    # io wrapper.
+    # This io wrapper is not compatible with the stdin= kwarg to .run - in that case
+    # it gets treated as a file like object and fails.
+    p = subprocess.run([cmd] + args[1:], capture_output=True, encoding="utf-8", env=env)
+    sys.stdout.write(p.stdout)
+    sys.stdout.flush()
+    sys.stderr.write(p.stderr)
+    sys.stderr.flush()
+    sys.exit(p.returncode)
+
+
+def run(cmd, args):
```

Comment:
We should fix up the parameters here - at the calling location, we are sending this:
`run("git", ["git, "gc"])`
And then this helper function always has to get throw away one of the redundant "git"s. We should just send a single array of commands, since that's what subprocess takes

Reply:
This got closer to keeping the convention of the subprocess API but stopped slightly short for some reason:
exec API: `run("git", ["git", "gc"])` <- where we were
current run function: `run("git", ["gc"])` <- where we are
subprocess API: `run(["git", "gc"])` <- why not just do this